pkg/execution/svccatadmission/internaldns.go

```diff
@@ -0,0 +1,147 @@
+			return nil, errors.Wrap(err, "error parsing InternalDNS spec")
+		}
+		if len(internalDNSSpec.Aliases) == 0 {
+			return nil, errors.Errorf("cannot process InternalDNS with empty aliases list")
+		}
+		domainsToCheck = make(chan string, len(internalDNSSpec.Aliases))
+		for _, alias := range internalDNSSpec.Aliases {
+			domainsToCheck <- alias.Name
+		}
+	default:
+		return nil, errors.Errorf("unsupported resource, got %v", admissionRequest.Resource)
+	}
+	close(domainsToCheck)
+
+	// fetching service data from service central
+	serviceName := getServiceNameFromNamespace(admissionRequest.Namespace)
+	serviceCentralData, err := getServiceData(ctx, scClient, serviceName)
+	if err != nil {
+		return nil, errors.Errorf("error fetching service central data for serviceName %q", serviceName)
+	}
+
+	// types to store parallel domain owner fetches
+	type domainOwnership struct {
+		requestedDomain         string
+		existingDomainAliasInfo *microsserver.AliasInfo
+		err                     error
+	}
+	domainsOwnership := make([]domainOwnership, cap(domainsToCheck))
+
+	// worker groups to parallel fetch requestedDomain owners
+	numWorkers := 5
+	wg := sync.WaitGroup{}
+	wg.Add(numWorkers)
+	var mutex = sync.Mutex{}
+	for i := 0; i < numWorkers; i++ {
+		go func() {
+			defer wg.Done()
+			for domain := range domainsToCheck {
+				aliasInfo, err := microsServerClient.GetAlias(ctx, domain)
+				if aliasInfo != nil {
+					logger.Infof("got micros server response %v for domain %v", aliasInfo, domain)
+				}
+				if err != nil {
+					logger.Errorf("got micros server error %v for domain %v", err, domain)
+				}
+				// concurrent safe append
+				func() {
+					mutex.Lock()
+					defer mutex.Unlock()
+					domainsOwnership = append(domainsOwnership, domainOwnership{
+						requestedDomain:         domain,
+						existingDomainAliasInfo: aliasInfo,
+						err:                     err,
+					})
+				}()
+			}
+		}()
+	}
+	wg.Wait()
+
+	// checking if domains are allowed to be migrated
+	for _, domainOwnership := range domainsOwnership {
+		if domainOwnership.err != nil {
+			logger.Errorf("error requesting alias info for %q from micros server", domainOwnership.requestedDomain)
+			return nil, errors.Wrapf(err, "error requesting alias info for %q from micros server", domainOwnership.requestedDomain)
+		}
+		if domainOwnership.existingDomainAliasInfo != nil {
+			// if we get a domain name with empty Service object from micros-server, it means is a domain provided via OSB for a micros 2 service
+			// in that case micros-server does not have ownership information, so we allow it and let the provider update logic deals with the situation
+			if domainOwnership.existingDomainAliasInfo.Alias.DomainName != "" && domainOwnership.existingDomainAliasInfo.Service.Owner == "" {
+				logger.Infof("domain %q returned no ownership information from micros server, so it is a domain provided via OSB, allowing", domainOwnership.requestedDomain)
```

Comment:
As I said before, the documentation says that in case of admission success, the reason is never consulted. So the only way to output messages for success cases is either through logs and comments (I'm doing both)